std/hash/poseidon2/poseidon2.go

```diff
@@ -0,0 +1,19 @@
+package poseidon2
+
+import (
+	"fmt"
+
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/hash"
+	poseidon2 "github.com/consensys/gnark/std/permutation/poseidon2"
+)
+
+// NewMerkleDamgardHasher returns a Poseidon2 hasher using the Merkle-Damgard
+// construction with the default parameters.
+func NewMerkleDamgardHasher(api frontend.API) (hash.FieldHasher, error) {
+	f, err := poseidon2.NewPoseidon2(api)
+	if err != nil {
+		return nil, fmt.Errorf("could not create poseidon2 hasher: %w", err)
+	}
+	return hash.NewMerkleDamgardHasher(api, f, 0), nil
```

Comment:
Is IV=0 temporary or it indeed doesn't have preimage?

Reply:
I think it should not be known, but I'll check. For MiMC we also use 0.

Reply:
I think it should be good, https://www.poseidon-initiative.info/ has set finding preimage of 0 as one of the challenges.